### ISSUE TITLE
Google deferred CSS vs loadCSS

### ISSUE THREAD

**Issue by nextgenthemes** [filamentgroup/loadCSS#244]
This is what Google has in their docs about CSS loading. Can you please explain to me how this compares to loadCSS and what the advantages and disadvantages between this two methods are?

```html
<html>
  <head>
    <style>
      .blue{color:blue;}
    </style>
    </head>
  <body>
    <div class="blue">
      Hello, world!
    </div>
    <noscript id="deferred-styles">
      <link rel="stylesheet" type="text/css" href="small.css"/>
    </noscript>
    <script>
      var loadDeferredStyles = function() {
        var addStylesNode = document.getElementById("deferred-styles");
        var replacement = document.createElement("div");
        replacement.innerHTML = addStylesNode.textContent;
        document.body.appendChild(replacement)
        addStylesNode.parentElement.removeChild(addStylesNode);
      };
      var raf = requestAnimationFrame || mozRequestAnimationFrame ||
          webkitRequestAnimationFrame || msRequestAnimationFrame;
      if (raf) raf(function() { window.setTimeout(loadDeferredStyles, 0); });
      else window.addEventListener('load', loadDeferredStyles);
    </script>
  </body>
</html>
```

**Comment by rsteinwand** [comment]
I still use Google's deferred CSS code on our public website. 

Google's code is a lot smaller, but LoadCSS knocks a second, maybe two off the page load time for heavy page, if implemented correctly. I'm still testing the pre-1.0 code, but have it on our home page and it's looking promising (and it's not blocking according to PSI).

**Comment by scottjehl** [comment]
To answer the original question: for CSS that doesn't need to load in a hurry, the code example above will work fine. It will start downloading CSS later than the loadCSS recommended workflow (which uses `rel=preload`) for two reasons:

1. The CSS in the snippet above is not referenced using a standard HTML link element, so it is not exposed to the browser's early parsing steps which would make the request very early in page processing. In particular, `link[rel=preload]`, which is now our recommended starting point for loading asynchronous CSS, will start downloading very early in the parsing steps.
2. Google's script approach waits for requestAnimationFrame before starting its request for the CSS. That is one way to safely ensure that the stylesheet will load without blocking rendering, but the first RAF callback can potentially happen late in the loading process depending on how many synchronous steps take place before the first animation frame renders.

So, in short, the starting HTML above is less optimal now that we can use `rel=preload`, and the request starts late due to different event listeners. (Comparing this directly to loadCSS.js, it still will start the download later, as loadCSS tries to start loading immediately rather than waiting for RAF).
